doc/programming/parsing.rst

```diff
@@ -1828,13 +1828,10 @@ control the current position:
     Returns iterator to the current input position in the stream fed
     into this unit.
 
-For random access, you'd typically get the current position through
-``input()``, subtract from it the desired number of bytes you want to
-back, and then use ``set_input`` to establish that new position. By
-further storing iterators as unit variables you can decouple these
-steps and, e.g., remember a position to later come back to.
-
-Here's an example that parses input data twice with different sub units:
+You can achieve random access by saving an iterator from ``input()``
+in a unit variable, then later return to that position by calling
```

Comment:
Yeah, this is good. The only thing is that now it seems to suggest that one *only* return to the position that `input()` returns. But moving the iterator forward (i.e., further into the unit), is fine and often more useful. How about this tweak to point that out:

```suggestion
in a unit variable, then later return to that position (or one derived from it) by calling
```